entity/machine.go

```diff
@@ -85,6 +93,53 @@ type Machine struct {
 	Locked                       bool                `json:"locked,omitempty"`
 }
 
+func (m *Machine) UnmarshalJSON(data []byte) error {
+	type machine Machine
+
+	if err := json.Unmarshal(data, (*machine)(m)); err != nil {
+		return err
+	}
+
+	// NOTE: deduce field value for backward compatibility
+	// `enable_hw_sync` field is exposed via API in MAAS 3.4+
+	// but hardware sync feature was added in 3.2
+	if !m.EnableHwSync {
+		t := MAASTime{}
+		if m.HwSyncInterval != 0 && (m.HwNextSync != t || m.HwLastSync != t) {
+			m.EnableHwSync = true
+		}
+	}
+
+	return nil
+}
+
+// MAASTime is a custom time format returned by MAAS API
+// which is not RFC3339
+type MAASTime struct {
+	time.Time
+}
+
+func (t *MAASTime) UnmarshalJSON(data []byte) error {
+	str, err := strconv.Unquote(string(data))
+	if errors.Is(err, strconv.ErrSyntax) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	temp, err := time.Parse("2006-01-02T15:04:05.999", str)
+	if err != nil {
+		return err
+	}
+	*t = MAASTime{temp}
+	return nil
+}
+
+// String returns MAASTime in RFC3339Nano format
+func (t *MAASTime) String() string {
```

Comment:
the receiver on this method doesn't need to be a pointer

Reply:
Nice catch!